### ISSUE TITLE
Getting my game to display

### ISSUE THREAD

**Issue by Shikamaru5** [facebookresearch/nle#234]
Hello,

I am using Docker to run the game and have been able to successfully run it for awhile, all was going well until I got the final touches done on my UI(AI) and connected it the docker image/container provided by the nle team. The issue was that when I run it, the operation completes but doesn't show me anything, the root workspace# thing doesn't show up, and the game doesn't appear. It just shows ready for the new command. I figured that the issue was that I wasn't environment rendering properly so trying to put new code in I have now been receiving messages informing me that it can't find the NetHackChallenge-v0.

I have the code as follows put in:

env = gym.make("NetHackChallenge-v0", savedir=None)  
env.reset()
env.step(action)

The program informs me with this error message:

 File "C:\Users\Kalvin\Desktop\Dockerfile\gym\envs\registration.py", line 131, in spec
    raise error.UnregisteredEnv('No registered env with id: {}'.format(id))
gym.error.UnregisteredEnv: No registered env with id: NetHackChallenge-v0

I also could just be wasting my time with trying to render the environment because it is already in the Docker image, so perhaps it's that I've somehow blocked the game from executing by successfully adding the algo file. Not entirely sure but any help would be appreciated.

I appreciate you reading this, thank you.

Sincerely Shikamaru5

**Comment by Shikamaru5** [comment]
Actually sorry the gym make stuff doesn't make sense because it wouldn't be running on a windows os so ya I'm confused to why the program doesn't show me what it's doing when it runs my algorithm. When I do control B to build it in my program it shows that it runs successfully and it doesn't give me any errors running in terminal but it doesn't give me any output.

Thanks again for everyone's patience, it's awesome working in a community that is willing to help out and give support.

Sincerely Shikamaru5

**Comment by Shikamaru5** [comment]
Ok, so I believe that I have figured out what the issue is, I think because I am in a Docker container and am trying to run multiple processes, the game, and then the AI, it only reads one or the other depending on which comes last in the file. I've searched for a number of solutions to this and thus far haven't come up with any success. I have been trying to work supervisord which I think might be the right way to go, however, it has come to my attention that I may need to make multiple Docker images that do the processes separately and try to link them. Whichever the solution, I have not come across an answer as of yet, any help would be appreciated.  
Currently my Dockerfile is pretty simple;
 I use the FROM fairnle/nle-focal, 
and then make a WORKDIR to a directory
RUN pip install my algorithm
COPY my algorithm ./my algorithm
ADD supervisord.conf /directory/supervisord/conf.d/supervisord.conf
COPY . .
ENTRYPOINT ["python", "my algorithm"]

thank you for reading.

Sincerely Shikamaru5
